### PR TITLE
tests passing for shared postcode example

### PR DESCRIPTION
**Pull request by goodviber**
### Context
we need to avoid calling postcode.io api if the postcode is not in the correct format initially
### Changes proposed in this pull request
1. update postcode_validator to check presence and format before calling api
2. change placement_suitability address_postcode to postcode to ease shared examples

### CODE REVIEW

File: db/migrate/20201215095226_change_address_postcode_to_postcode.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class ChangeAddressPostcodeToPostcode < ActiveRecord::Migration[6.0]
+  def change
+    rename_column :placement_suitabilities, :address_postcode, :postcode
```
Comment (novotnyjakub) [inline db/migrate/20201215095226_change_address_postcode_to_postcode.rb:3]: Why this change? Wouldn't it be better to keep all the address fields with consistent naming?

**Review by novotnyjakub — APPROVED** [review]
👍  The only thing I wouldn't do is the column rename, as it falls out of consistency with the rest of the address fields. Otherwise it looks good and i like the abstraction to `PostcodeValidator`!

File: db/migrate/20201215095226_change_address_postcode_to_postcode.rb (hunk 1 above)
Comment (goodviber) [reply db/migrate/20201215095226_change_address_postcode_to_postcode.rb:3]: It was just to make it easier to use the shared example. We could update the naming on the placement_needs to use the address prefix, I'm easy...
